Sources/TuistLoader/Loaders/RecursiveManifestLoader.swift

```diff
@@ -68,19 +70,33 @@ public class RecursiveManifestLoader: RecursiveManifestLoading {
 
             let project = try manifestLoader.loadProject(at: path)
             cache[path] = project
-            paths.append(contentsOf: try dependencyPaths(for: project, path: path))
+            paths.append(contentsOf: try dependencyPaths(for: project, path: path, dependencies: dependencies))
         }
 
         return LoadedProjects(projects: cache)
     }
 
-    private func dependencyPaths(for project: Project, path: AbsolutePath) throws -> [AbsolutePath] {
+    private func dependencyPaths(
+        for project: ProjectDescription.Project,
+        path: AbsolutePath,
+        dependencies: DependenciesGraph
+    ) throws -> [AbsolutePath] {
         let generatorPaths = GeneratorPaths(manifestDirectory: path)
         let paths: [AbsolutePath] = try project.targets.flatMap {
             try $0.dependencies.compactMap {
                 switch $0 {
                 case let .project(target: _, path: projectPath):
                     return try generatorPaths.resolve(path: projectPath)
+                case let .thirdParty(name):
+                    guard let dependency = dependencies.thirdPartyDependencies[name] else {
+                        return nil
+                    }
+
+                    switch dependency {
+                    // TODO: generate project for source based dependencies and return its Path
+                    case .xcframework:
+                        return nil
+                    }
```

Comment:
This component may not be suitable for this sort of action 🤔

The manifest loader currently only deals with loading the manifest contents from disk - the `TuistKit.Generator` could use the loaded information to perform additional transformations or side effects if needed as another step in the "pipeline" so to speak.

https://docs.tuist.io/contributors/generation-pipeline

Reply:
Just to be sure I understood correctly, you would:
- generate the project earlier (e.g. in the `Generator.swift` before calling the `RecursiveManifestLoading`
- in the `RecursiveManifestLoading` we would just change the logic to also load the "generated projects", as if it would be a `project` dependency

is it correct?

Reply:
The generation of the Project.swift for the SPM dependencies might even be moved to the `tuist dependencies fetsch/update` 🤔 
WDYT @fortmarek @laxmorek ?

Reply:
> The generation of the Project.swift for the SPM dependencies might even be moved to the tuist dependencies fetsch/update

I think that's a great suggestion - one thing that we should probably be thoughtful about is how this will behave when you update tuist and we introduce some breaking changes. `tuist dependencies fetch/update` might not be run so often, so I'd expect it to be consistent through updates or letting users know that they should update the dependencies. Alternatively, we could try to regenerate the manifests ourselves if we can detect when it's necessary. If we can generate the manifests outside of `tuist generate`, that'd also be good for the performance of that command.

Reply:
also why do you need to pass `dependencies` through all those functions? If this is the only reason, maybe we could delete those?

Reply:
> Just to be sure I understood correctly, you would:
>
>  - generate the project earlier (e.g. in the Generator.swift before calling the RecursiveManifestLoading
> - in the RecursiveManifestLoading we would just change the logic to also load the "generated projects", as if it would be a project dependency

Ah sorry, I meant we wouldn't need to change the `RecursiveManifestLoader` at all seeing it's probably not the best place to do this sort of logic, but instead move any logic relating to third party dependencies elsewhere as you and @fortmarek suggested _(best to be independent from `tuist generate` if at all possible)_

Reply:
I agree in moving the generation logic somewhere else, but loading the generated project (which at that point would be a "normal" tuist project, should be done here, isn't it?

Reply:
I have updated the logic, let me know what you think 🙌